src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -672,24 +748,17 @@ async function getFlows(
   // - Fills that are considered "invalid" by the spoke pool client.
   const fills: UbaFlow[] = (
     await Promise.all(
-      spokePoolClient.getFills().map(async (fill: FillWithBlock): Promise<UbaFlow | undefined> => {
-        const validWithinBounds =
-          fill.repaymentChainId === spokePoolClient.chainId &&
-          fill.fillAmount.eq(fill.totalFilledAmount) &&
-          fill.updatableRelayData.isSlowRelay === false &&
-          fill.blockNumber >= (fromBlock as number) &&
-          fill.blockNumber <= (toBlock as number);
-        if (!validWithinBounds) {
-          return undefined;
-        }
-        const matchingDeposit = await resolveCorrespondingDepositForFill(fill, spokePoolClients);
-        if (matchingDeposit === undefined) {
-          return undefined;
-        }
-        return {
-          ...fill,
-          quoteBlockNumber: matchingDeposit.quoteBlockNumber,
-        };
+      (
+        await getFills(chainId, hubPoolClient, spokePoolClients, {
+          fromBlock,
+          toBlock,
+          repaymentChainId: chainId,
+          isSlowRelay: false,
+        })
+      ).filter((fill: FillWithBlock) => {
+        // We only want to include full fills as flows. Partial fills need to request refunds and those refunds
+        // will be included as flows.
```

Comment:
Could we use `filterAsync`

Reply:
this is a `sync` filter actually. The `array` is `await getFills` and the filter function is syncrhonous